### ISSUE TITLE
Run in Terminal Button doesn't work

### ISSUE THREAD

**Issue by ArianArsenal** [oleg-shilo/cs-script.vscode#39]
after executing the Run in Terminal button the terminal opens up and runs the code correctly, but after killing the terminal the button in the right corner (Run in Terminal) doesn't open the terminal again unless you close the vs code and open it again.

**Comment by oleg-shilo** [comment]
Yes. It is a defect. Thank you form reporting.
Fixed now. PLease update the extension.

**Comment by ArianArsenal** [comment]
> Yes. It is a defect. Thank you form reporting. Fixed now. PLease update the extension.

I updated the extension and now the Run button doesn't work and it says CS-Script is busy
also the run in terminal button now gives me an error which I didn't get in the previous version 

A fatal error was encountered. The library 'hostpolicy.dll' required to execute the application was not found in 'C:\User\AppData\Roaming\Code\User\cs-script.user\dotnet\'.
Failed to run as a self-contained app.
  - The application was run as a self-contained app because 'C:\User\AppData\Roaming\Code\User\cs-script.user\dotnet\cscs.runtimeconfig.json' was not found.
  - If this should be a framework-dependent app, add the 'C:\User\AppData\Roaming\Code\User\cs-script.user\dotnet\cscs.runtimeconfig.json' file and specify the appropriate framework.

**Comment by oleg-shilo** [comment]
Yes, it is a problem it is not caused by the extension but by the CS-SCript itself.

The error "CS-Script is busy" means that you have started the launcher but it does hang for some reason.
You can just kill that instance by using this command:
![image](https://github.com/user-attachments/assets/32c19e67-abb9-4051-b77a-24a2fc792f06)
It's not a solution but only the way to resume your VSCode state.

Can you check, what is the deployment of CS-Script that you are using? I speculate that after upgrade the cs-script config got changed accidentally.

What is your output execution output? 
THis is what I have:
![image](https://github.com/user-attachments/assets/fdff1b67-051b-44e5-81f2-00121bf64f29)

**Comment by oleg-shilo** [comment]
And also can you just check the version of your cs-script deployment:
```
css --version
# or
dotnet "<whatever your path>\cscs.dll" --version
```

**Comment by ArianArsenal** [comment]
This is my output 
![image](https://github.com/user-attachments/assets/1f9b4b02-21df-4f08-a364-2138d5f8d058)

which i also the same result as the deployment version check
![image](https://github.com/user-attachments/assets/5ddcfd4a-2e37-4e4c-94d3-031eea331554)

**Comment by oleg-shilo** [comment]
Yep, this means that your cs-script deployment is corrupted.  Most likely you are missing this file:

![image](https://github.com/user-attachments/assets/7be383e8-cfc3-40d0-8124-a65624c1419d)

We will investigate why but let's fix your environment first.

I suggest you install CS-Script outside of the extension. It's more flexible and just easier to update when a new version of CS-Script is released.

You can install CS-SCript from Chocolatey:
```
sudo choco install cs-script
```

After that, you can instruct the extension to use this new deployment of cs-script:

![image](https://github.com/user-attachments/assets/4e6ff91d-4087-4e94-96f6-73e2722a68ca)

**Comment by oleg-shilo** [comment]
And I managed to reproduce the problem by manually deleting that file.
The problem is that the extension does not have a command for redeploying the built-in engine. This is a design limitation and I am going to address it.

But you can still trigger the redeployment manually by deleting the version file:
![image](https://github.com/user-attachments/assets/9ebc0e90-d01f-4da1-9b02-e95175de0f7f)

Though I would still prefer the choco deployment

**Comment by ArianArsenal** [comment]
Thanks , i manually installed the CS-Script and changed the path to that , it's fixed for now so thank you.

**Comment by oleg-shilo** [comment]
Great.
The deployment is not transactional. It cannot be. The VSCode can crash in the middle of the deployment when only half of the files are installed. So actually not having a clear re-deployment option is an unnecessary constraint.
So I am going to implement it in the next release.
